### ISSUE TITLE
Commenting text with headers and bullets does not comment all content

### ISSUE THREAD

**Issue by santtu** [typora/typora-issues#4075]
If I have a text like

```
# Header

Some text here.

* Bullet
* Another bullet

More text.
```

and select the text and use the comment command (Format>Comment), it will produce the the following text:

```
# <!--Header-->

<!--Some text here.-->

* <!--Bullet-->
* <!--Another bullet-->

<!--More text.-->
```

The problem is that it leaves the header (empty `#`) and bullets intact instead of commenting them, resulting in any rendering empty header and empty bullets to appear.  So, in a way, commenting out a block of content does not comment it entirely out.

I would expect the commenting to make all of the *content* into comments, and not just the text, i.e.:

```
<!--# Header-->

<!--Some text here.-->

<!--* Bullet-->
<!--* Another bullet-->

<!--More text.-->
```

**Comment by abnerlee** [comment]
I think this can be covered by #1289 
```markdown
<!--# Header

Some text here.

* Bullet
* Another bullet

More text.-->
```

seems more cleaner

**Comment by santtu** [comment]
@abnerlee umm how come? The problem I described is not about how Typora renders comments, but how the Comment menu command works.

**Comment by abnerlee** [comment]
I would think user would prefer 
```
<!--# Header

Some text here.

* Bullet
* Another bullet

More text.-->
```
as the comment menu command output.

using `<!-- -->` will leave out empty white spaces between paragraphs, which may also affects the final output.

**Comment by santtu** [comment]
I agree with commenting out all of the marked area, which requires #1289, but the **menu command "Comment"** is not addressed by #1289 itself.
